generators/app/templates/.gitignore

```diff
@@ -1,3 +1,35 @@
+################################################
+# macOS
```

Comment:
This gitignore contains a lot of things I don't think I've ever once had to deal with in my life as a developer, but is also missing quite a few things that we have to deal with all the time. Here's a git ignore from a recent project, for instance.

```
dist/
craft/storage/*

# MISC FILES
# ----------------------------------
.DS_Store
.sass-cache
.env

# PACKAGE MANAGERS
# ----------------------------------
bower_components
npm-debug.log
node_modules

# GENERATED ASSETS
# ----------------------------------
public/assets/**/_*
public/scripts
public/styles
public/images
public/imager/**/*
public/assets
craft/rev-manifest.json
```

I'm not sure whether we should err on the side of including too much or too little, but this feels to me like it is on the _too much_ side right now.